### PR TITLE
PE-3710: Profile side bar on mobile

### PR DESCRIPTION
**Pull request by matibat**
Changes:
1. The view for Desktop hasn't changed
2. The view for Mobile...
  a. now occupies the full height
  b. has now an X button
  c. has the bottom of the view filled with the same color as the Logout button
  d. won't have the border radius of the overlay card

--- Releases ---
Android release: https://appdistribution.firebase.google.com/testerapps/1:305132849030:android:6cf0cd5ec064fad3ffce07/releases/55fqii2oanli0
iOS release: https://appdistribution.firebase.google.com/testerapps/1:305132849030:ios:06ea33a95a2e0d42ffce07/releases/27f35rmmro258

### CODE REVIEW

**Comment by matibat** [comment]
https://github.com/ardriveapp/ardrive-web/assets/22564541/fad73173-59b9-4a5f-9907-17ac5a8fc3a5

File: lib/components/profile_card.dart [hunk 1]
```diff
@@ -61,43 +63,85 @@ class _ProfileCardState extends State<ProfileCard> {
   }
 
   Widget _loggedInView(BuildContext context) {
+    return ArDriveClickArea(
+      child: ScreenTypeLayout.builder(
```
Comment (matibat) [inline lib/components/profile_card.dart:67]: **For the reviewer -**  Now conditionally rendering the overlay depending on the platform

File: lib/components/profile_card.dart [hunk 2]
```diff
@@ -61,43 +63,85 @@ class _ProfileCardState extends State<ProfileCard> {
...
+          setState(() {
+            _showProfileCard = false;
+          });
+        }
+      },
+      visible: _showProfileCard,
+      anchor: Aligned(
+        follower: Alignment.topRight,
+        target: Alignment.bottomRight,
+        offset: isMobile ? const Offset(12, -60) : const Offset(0, 4),
```
Comment (matibat) [inline lib/components/profile_card.dart:99]: **For the reviewer -** The offset of the anchor is now set, in the case of the mobile platform, to be aligned with the top-right corner of the screen.

File: lib/components/profile_card.dart [hunk 3]
```diff
@@ -61,43 +63,85 @@ class _ProfileCardState extends State<ProfileCard> {
...
     );
   }
 
   Widget _buildProfileCardContent(
     BuildContext context,
     ProfileLoggedIn state,
-    String walletAddress,
-  ) {
+    String walletAddress, {
+    bool isMobile = false,
```
Comment (matibat) [inline lib/components/profile_card.dart:115]: I'd prefer to have it as required
```suggestion
    required bool isMobile,
```

File: lib/components/profile_card.dart [hunk 4]
```diff
@@ -61,43 +63,85 @@ class _ProfileCardState extends State<ProfileCard> {
...
+  }) {
     return ArDriveCard(
       contentPadding: const EdgeInsets.all(0),
       width: 281,
+      height: isMobile ? double.infinity : null,
```
Comment (matibat) [inline lib/components/profile_card.dart:120]: **For the reviewer -** for mobile, the height will be the screen size

File: lib/components/profile_card.dart [hunk 5]
```diff
@@ -61,43 +63,85 @@ class _ProfileCardState extends State<ProfileCard> {
...
+      borderRadius: isMobile ? 0 : null,
```
Comment (matibat) [inline lib/components/profile_card.dart:121]: **For the reviewer -** the border radius will be disabled for mobile, and the default one for desktop.

File: lib/components/profile_card.dart [hunk 6]
```diff
@@ -61,43 +63,85 @@ class _ProfileCardState extends State<ProfileCard> {
...
       boxShadow: BoxShadowCard.shadow60,
       content: Column(
         mainAxisSize: MainAxisSize.min,
         crossAxisAlignment: CrossAxisAlignment.start,
         children: [
+          if (isMobile)
```
Comment (matibat) [inline lib/components/profile_card.dart:127]: **For the reviewer -** this is the X button

**Review by matibat — COMMENTED** [review]
Self-review chackpoint

**Review by karlprieb — APPROVED** [review]

**Review by thiagocarvalhodev — DISMISSED** [review]
LGTM!

**Review by thiagocarvalhodev — APPROVED** [review]
LGTM

**Pull request merged**